enhancements/sig-policy/89-operator-policy-kind/README.md

```diff
@@ -0,0 +1,557 @@
+but multiple policies are often required, and these require the policy author to have a deep
+understanding of the OLM resources being managed. For example, the policy author might need to
+consider whether an OperatorGroup must be created, and incorrectly creating multiple OperatorGroups
+can cause operator installations to fail. If the policy author wants to monitor the health of the
+operator, they would need to create policies to monitor things like the status of the Subscription
+(which can not necessarily be incorporated into the policy to create the Subscription), and related
+resources which the operator manages like Deployments. Upgrading or uninstalling the operator would
+require additional policies, and the status feedback from `ConfigurationPolicies` might not provide
+enough useful information when things do not go according to plan in these scenarios, requiring
+manual intervention from an admin.
+
+A new controller can handle the relationships between resources better, for example knowing when an
+OperatorGroup should be created, or identifying the Deployments and CustomResourceDefinitions
+managed by a ClusterServiceVersion. This information could be incorporated into the policy status,
+helping administrators understand the status of the operators from the hub cluster. The new kind can
+also provide new controls over the upgrades and uninstallation of these operators, which would be
+difficult with current solutions.
+
+### Goals
+
+1. It should be easy to install an operator on a managed cluster with just one policy.
+2. It should be possible to specify what upgrades should be allowed performed on an operator.
+3. The policy status should report on the health of the operator, both the OLM resources like the
+   Subscription, as well as the associated Deployments.
+4. The policy should report when the CatalogSource for the operator is unhealthy. It should be
+   configurable whether this will cause the policy to become NonCompliant.
+5. What happens when the policy is deleted should be configurable: the operator might be deleted, or
+   the operator and its CRDs might be deleted, or nothing might be deleted. By default, only the
+   Subscription and CSVs will be deleted (matching the suggestion from OLM).
+
+### Non-Goals
+
+1. The installation of OLM on managed clusters will not be automated by this enhancement.
+2. Management of "Operands" will not be included in this new type - they should be defined in
+   separate `ConfigurationPolicies`, possibly with a
+   [dependency](../74-order-policy-execution/README.md) on the `OperatorPolicy`. ("Operand" here
+   refers to resources that would be created by a user, and managed by the operator, eg instances of
+   CRDs which are defined by the operator.)
+3. Creation of `CatalogSources` will not be included in the new type - they should be defined in
+   separate `ConfigurationPolicies`.
```

Comment:
I agree that the catalog source should be handled in a different policy. Just to highlight that currently the catalog source API does not provide `observerdGeneration` in its status. Any thing we derive from its status can be stale if its spec just got updated. This can make the policy ordering ineffective. The operator policy can go/stay compliant when it shouldn't be. Same concern goes with the `catalogSourceUnhealthy` field in `statusConfig`. It wouldn't do its job if we set it to `NonCompliant` in cases like this. I opened an RFE for this but it is set to lowest priority possible ...
https://issues.redhat.com/browse/RFE-3846

I've linked it to this enhancement JIRA. If we can push for it together and get it implemented by OLM, this enhancement will be more successful :)